File: Data Structures & Algorithms/is-palindrome/submission-0.js
class Solution {
    isAlphanumeric(char) {
        return (
            (char >= 'a' && char <= 'z') ||
            (char >= 'A' && char <= 'z') ||
            (char >= '0' && char <= '9')
        );
    } 

    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        let str = "";

        for (const c of s) {
            if (this.isAlphanumeric(c)) {
                str += c.toLowerCase();
            }
        }

        let left = 0;
        let right = str.length - 1;

        while (left < right) {
            if (str[left] !== str[right]) return false;
            left++;
            right--;
        }

        return true;
    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p is None and q is None:
            return True
        elif (p is None and q) or (q is None and p):
            return False
        
        pTree = self.createLevelOrderArray(p)
        print(pTree)
        qTree = self.createLevelOrderArray(q)
        print(qTree)

        return pTree == qTree
    
    def createLevelOrderArray(self, tree):
        queue = deque([tree])
        treeArr = []
        treeArr.append(tree.val)

        while queue:
            node = queue.popleft()

            if node.left:
                queue.append(node.left)
                treeArr.append(node.left.val)
            else:
                treeArr.append(None)

            if node.right:
                queue.append(node.right)
                treeArr.append(node.right.val)
            else:
                treeArr.append(None)
        
        return treeArr
